cetmix_tower_server/tests/test_plan.py

```diff
@@ -1261,3 +1261,503 @@ def test_plan_with_another_plan_with_all_not_executable_condition(self):
+        6. Ensure the newly created variable value is properly linked to the plan
+         line action.
+        7. Validate that user_bob can delete  own variable value from a plan line
+           action he created.
+        """
+        # Remove User_bob from all groups
+        self.remove_from_group(
+            self.user_bob,
+            [
+                "cetmix_tower_server.group_user",
+                "cetmix_tower_server.group_manager",
+                "cetmix_tower_server.group_root",
+            ],
+        )
+        # Assign variable value to plan action
+        variable_value = self.env["cx.tower.variable.value"].create(
+            {
+                "variable_id": self.variable_os.id,
+                "value_char": "Windows 2k",
+                "plan_line_action_id": self.plan_line_1_action_1.id,
+            }
+        )
+
+        # group_user:
+        # Plan not assigned to server, ensure access to variable value in plan line
+        # action
+        # Add user to group
+        self.add_to_group(self.user_bob, "cetmix_tower_server.group_user")
+        self.write_and_invalidate(self.plan_1, **{"access_level": "1"})
+        test_plan_line_action_as_bob = self.plan_line_1_action_1.with_user(
+            self.user_bob
+        )
+        plan_line_action_vv_read_result = (
+            test_plan_line_action_as_bob.variable_value_ids.read([])
+        )
+        self.assertEqual(
+            plan_line_action_vv_read_result[0]["value_char"],
+            variable_value.value_char,
+            msg="Value should be the same",
+        )
+        # User not subscribed to the server: ensure no access to variable value in
+        # plan line action
+        self.write_and_invalidate(
+            self.plan_1, **{"server_ids": [(6, 0, [self.server_test_1.id])]}
+        )
+        with self.assertRaises(AccessError):
+            plan_line_action_vv_read_result = (
+                test_plan_line_action_as_bob.variable_value_ids.read([])
+            )
+        # User subscribed to the server: ensure access to variable value in
+        # plan line action
+        self.server_test_1.message_subscribe([self.user_bob.partner_id.id])
+        plan_line_action_vv_read_result = (
+            test_plan_line_action_as_bob.variable_value_ids.read([])
+        )
+        self.assertEqual(
+            plan_line_action_vv_read_result[0]["value_char"],
+            variable_value.value_char,
+            msg="Value should be the same",
+        )
+        # group_manager:
+        self.add_to_group(self.user_bob, "cetmix_tower_server.group_manager")
+        self.write_and_invalidate(self.plan_1, **{"access_level": "2"})
+        # User subscribed to the server: ensure user_bob cannot remove variable value
+        # from action line if he is not creator of action line
+        with self.assertRaises(AccessError):
+            test_plan_line_action_as_bob.write({"variable_value_ids": [(5, 0, 0)]})
```

Comment:
This doesn't remove the variable value, but removes all variable values from the x2m field. 
Although in this case it's supposed to be the same, yet we need to use more explicit way of doing that.
Eg what if there are more than one value and error is raised because of those extra ones?

Reply:
Fixed